### ISSUE TITLE
Getting `Error (Xcode): Linker command failed with exit code 1 (use -v to see invocation)` on Xcode 15.0.1 

### ISSUE THREAD

**Issue by Ariflo** [AstroxNetwork/agent_dart#71]
👋 We had [a couple of our Flutter SDK developers](https://github.com/magiclabs/magic-flutter/issues/45) bring up the `parseErrorData` issue https://github.com/AstroxNetwork/agent_dart/issues/70 and ask us to upgrade to the latest version of `agent_dart: ^1.0.0-dev.18` to bypass the error. 

Unfortunately, doing so leads to the following [error with Xcode](https://developer.apple.com/forums/thread/738672?login=true&page=1#771390022):  

![Screenshot 2023-11-06 at 5 06 48 PM](https://github.com/AstroxNetwork/agent_dart/assets/13407884/80b808e8-d555-426e-8a5b-0f407ccf0161)

Here's how the error looks like on my terminal 👇 :
```
Error (Xcode): Linker command failed with exit code 1 (use -v to see invocation)


Could not build the application for the simulator.
Error launching application on iPhone 15 Pro Max.
```

When I attempt to add `CoreAudioTypes.framework` to the flutter project inside of Xcode I'm met with this warning: `CoreAudioTypes.framework is not supported on Mac Catalyst`. 🤔 

![Screenshot 2023-11-06 at 5 08 57 PM](https://github.com/AstroxNetwork/agent_dart/assets/13407884/17f58cdb-ce08-481d-a971-f31590369684)

We've tried every which way to bypass the above error:
- Removed Mac Catalyst support from all dependencies
- Added the Audio Types to `agent_dart` directly vs the Runner target 
etc etc. 

As it stands we are unable to run [our demo flutter app](https://github.com/magiclabs/magic-flutter/tree/main/magic_demo) on iOS (Android works fine 😌)  with `agent_dart: ^1.0.0-dev.18` and switching back to the older version brings back issue https://github.com/AstroxNetwork/agent_dart/issues/70. Please help 🙏

**Comment by AlexV525** [comment]
Your provided log does not contain useful information. But the Xcode provides "Minimal IPHONEOS_DEPLOYTMENT_TARGET` mismatch" problem. Could you fix that before proceed? Also, click on "Undefined symbols" should help. And they don't sound like related to our packages.

**Comment by Ariflo** [comment]
@AlexV525 apologies, I forgot to mention that `pod install` also returns this error when attempting to run the command:

```
Analyzing dependencies
  End-of-central-directory signature not found.  Either this file is not
  a zipfile, or it constitutes one disk of a multi-part archive.  In the
  latter case the central directory and zipfile comment will be found on
  the last disk(s) of this archive.
unzip:  cannot find zipfile directory in one of agent_dart-v1.0.0.zip or
        agent_dart-v1.0.0.zip.zip, and cannot find agent_dart-v1.0.0.zip.ZIP, period.```

**Comment by Ariflo** [comment]
> ... But the Xcode provides "Minimal IPHONEOS_DEPLOYTMENT_TARGET mismatch" problem. Could you fix that before proceed?

![Screenshot 2023-11-07 at 4 43 05 PM](https://github.com/AstroxNetwork/agent_dart/assets/13407884/e4dc61e4-a561-4ad7-8f19-3ccef11fa505)

**Comment by Ariflo** [comment]
> click on "Undefined symbols" should help. 


It doesn't do anything 😢 
![Screenshot 2023-11-07 at 4 44 41 PM](https://github.com/AstroxNetwork/agent_dart/assets/13407884/264cf298-c0d3-4333-8ad3-84e30a5fadd1)

**Comment by AlexV525** [comment]
@Ariflo Could you depend on 2b4c8b9 and try again?

```yaml
dependencies:
  agent_dart:
    git:
      url: https://github.com/AstroxNetwork/agent_dart
      ref: 2b4c8b9
```

**Comment by Ariflo** [comment]
> @Ariflo Could you depend on [2b4c8b9](https://github.com/AstroxNetwork/agent_dart/commit/2b4c8b975dc12727f85ad2ef3d42f3812a156d8d) and try again?
> 
> ```yaml
> dependencies:
>   agent_dart:
>     git:
>       url: https://github.com/AstroxNetwork/agent_dart
>       ref: 2b4c8b9
> ```

That looks to have resolved my issue, thanks @AlexV525

**Comment by Ariflo** [comment]
@AlexV525 unfortunately pub.dev doesn't allow git dependencies in published packages. 😢  Let us know as soon as [the commit](https://github.com/AstroxNetwork/agent_dart/commit/2b4c8b975dc12727f85ad2ef3d42f3812a156d8d) is released along side a new version of your [flutter_secp256r1 package](https://github.com/AstroxNetwork/flutter_secp256r1) so that we may subsequently release our package and unblock our devs. Thanks again!

**Comment by AlexV525** [comment]
v1.0.0-dev.19 has been published.

**Comment by Ariflo** [comment]
@AlexV525 Unfortunately I'm still seeing the same problem with `agent_dart: 1.0.0-dev.19` installed. What's even more peculiar is that when I point to the repo's latest `git` ref, my app runs fine. 🤔 

Is it possible that the code pushed to pub.dev differs from the on the git repo? Any idea?

**Comment by AlexV525** [comment]
Should be fixed by v1.0.0-dev.20.

**Comment by Ariflo** [comment]
Works great, thanks again @AlexV525 🙇
